tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/PoPTests.cs

```diff
@@ -281,6 +283,64 @@ public async Task PopTestWithRSAAsync()
                  result).ConfigureAwait(false);
         }
 
+        [TestMethod]
+        public async Task PopTest_ExternalWilsonSigning_Async()
+        {
+            var confidentialApp = ConfidentialClientApplicationBuilder
+                .Create(PublicCloudConfidentialClientID)
+                .WithExperimentalFeatures()
+                .WithAuthority(PublicCloudTestAuthority)
+                .WithClientSecret(s_publicCloudCcaSecret)
+                .Build();
+
+            // Create an RSA key Wilson style (SigningCredentials)
+            var key = CreateRsaSecurityKey();
+            var popCredentials = new SigningCredentials(key, SecurityAlgorithms.RsaSha256);
+
+            var popConfig = new PoPAuthenticationConfiguration() 
+            {                 
+                PopCryptoProvider = new SigningCredentialsToPopCryptoProviderAdapter(popCredentials, true),                
+                DoNotSignHttpRequest = true,
+            };
+
+            var result = await confidentialApp.AcquireTokenForClient(s_keyvaultScope)
+                .WithProofOfPossession(popConfig)
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
+            Assert.AreEqual("pop", result.TokenType);
+            Assert.AreEqual(
+                TokenSource.IdentityProvider, 
+                result.AuthenticationResultMetadata.TokenSource);
+
+            SignedHttpRequestDescriptor signedHttpRequestDescriptor =
```

Comment:
This is using Wilson library to create the outer envelope

Reply:
Is this included in the future Wilson version? Since it can't find reference right now.

Reply:
No, no it's there. I forgot to push the changes to the csproj. I have now.

`<PackageReference Include="Microsoft.IdentityModel.Protocols.SignedHttpRequest" Version="6.15.0" />`